### ISSUE TITLE
Feature request: generator functions

### ISSUE THREAD

**Issue by dnsmkl** [mzipay/Autologging#3]
What do you think about generator functions? (i.e. logging yielded values)
Any plans of adding special support for it?
If not, would you accept pull request in that direction?

Demo of current behavior and modified behavior
```
import logging
import sys
from autologging import TRACE, traced, logged
logging.basicConfig(
    level=TRACE, stream=sys.stdout,
    format="%(levelname)s:%(name)s:%(funcName)s:%(message)s")

@traced
def gen_func(x, y):
    yield x + y
    yield x + y + 1


for a in gen_func(7, 9):
    pass
```
Current ouput is:
```
TRACE:__main__:gen_func:CALL *(7, 9) **{}
TRACE:__main__:gen_func:RETURN <generator object gen_func at 0x012EAEA8>
```

Alternative output - 1 (trace generator object in log just as any other object ):
```
TRACE:__main__:gen_func:CALL *(7, 9) **{}
TRACE:__main__:gen_func:RETURN <generator object gen_func at 0x012EAEA8>
TRACE:__main__.generator_gen_func:__next__:RETURN 16
TRACE:__main__.generator_gen_func:__next__:RETURN 17
```

Alternative output - 2 (trace generator func in log as special case):
```
TRACE:__main__:gen_func:GENERATOR_FUNCTION_CALL *(7, 9) **{}
TRACE:__main__:gen_func:RETURN <generator object gen_func at 0x012EAEA8>
TRACE:__main__:gen_func:YIELD 16
TRACE:__main__:gen_func:YIELD 17
```

**Comment by mzipay** [comment]
I love this idea. I had already scrapped an approach to doing it that went too far down the proverbial rabbit hole... I would absolutely accept a pull request.
Looking at the examples, I prefer "Alternative output -2" because it uses the actual function name as written in the source; that makes log parsing (either visually or programmatically) more intuitive.

**Comment by mzipay** [comment]
Support for tracing generator iterators was just added in commit c48e11eb62095b2765a4211c03367899930dce68. The CALL/RETURN tracing remains, but there is now additional YIELD/STOP tracing when the return value from a function is a generator. See the example code in the project README.
(Note: **only** generators are traced with this enhancement; this means if you do something like directly return `enumerate(something)`, `reversed(something)`, `iter(something)`, etc. those will _not_ be automatically traced.)

**Comment by mzipay** [comment]
The YIELD/STOP tracing will be included in release 1.2.0 along with default __call__ tracing (from issues/4). Master branch is currently 1.2.0b0.
